mbuild/utils/specific_ff_to_residue.py

```diff
@@ -0,0 +1,385 @@
+import os
+import mbuild as mb
+import parmed as pmd
+
+from mbuild.compound import Compound
+from warnings import warn
+from xml.dom import minidom
+from mbuild.utils.io import has_foyer
+
+
+def specific_ff_to_residue(structure,
```

Comment:
Just something that is crossing my mind. I think after we merge this into mbuild, and before we decide to merge into gmso, we should consider converting this into a plugin, or maybe just an entire package itself. The amount of logic needed is quite substantial and would probably be better served as a sort of `mosdef-gomc` package. Especially since there is a lot of necessary information needed for GOMC. We could still have the base information for the CHARMM filetypes, but more specific GOMC utilities would be best represented in a separate repository.

Reply:
I'm not entirely sure you can decouple this logic, or similar residue name addition from CHARMM, since CHARMM deals with proteins, which have unique residues.  The current logic only supports a residue is a molecule (GOMC style), and does not support many residues per molecule (a protein).  I agree this will need to be considered in the future implementations.  My initial thought is just to be able to pass the residues from the mol2 ... , into mbuild and thru to the final pdb and psf files without requiring this "mbuild/utils/specific_ff_to_residue.py file".  We definitely should have a discussion on the best way to handle this in the future, upon the integration with GMSO.